src/lib.rs

```diff
@@ -430,6 +445,21 @@ impl SecretKey {
         Ok(SecretKey::from_mut(&mut fr))
     }
 
+    /// Deserialize a hex-encoded representation of a `SecretKey` to a `SecretKey` instance.
+    pub fn from_hex(hex: &str) -> Result<Self> {
+        let sk_bytes = hex::decode(hex).map_err(Error::HexConversionFailed)?;
+        let sk_bytes: [u8; SK_SIZE] = match sk_bytes.try_into() {
+            Ok(bytes) => bytes,
+            Err(_) => return Err(Error::InvalidBytes),
+        };
```

Comment:
Again here, can convert this to a one-liner, map the error and `?` to return early.